pwa-devdocs/src/tutorials/cloud-deploy/index.md

```diff
@@ -0,0 +1,220 @@
+
+If it does not exist, [create a `package.json`][] file in your project directory.
+
+```sh
+npm init -y
+```
+
+Add your PWA Studio storefront to the list of dependencies in this file.
+
+```json
+{
+    "dependencies": {
+        "@magento/venia-concept": "~2.0.0"
+    }
+}
+```
+
+## Add Composer dependencies
+
+Add the repository information for the UPWARD PHP server and UPWARD Connector to the `repositories` section in your `composer.json`.
+
+```json
+"upward-connector": {
+    "type": "vcs",
+        "url": "https://github.com/magento-research/magento2-upward-connector"
+
+},
+  "upward-php": {
+      "type": "vcs",
+          "url": "https://github.com/magento-research/upward-php"
+
+  }
+```
+
+To have composer install these packages, add them to the `require` section of the `composer.json` file.
+
+```json
+"magento/module-upward-connector": "^1.0.0",
+"magento/upward": "^1.0.0"
+```
+
+## Add Venia sample data (optional)
+
+The Venia storefront works best with the Venia sample data installed. There is an [automated script](https://magento-research.github.io/pwa-studio/venia-pwa-concept/install-sample-data/) in the `@magento/venia-concept` package, or you can follow the manual steps here.
+
+If you are deploying your own custom storefront, you may skip this step and continue to the next section.
+
+### Add sample data repository information
+
+Add the Venia sample data repository information to the `repositories` section of the `composer.json` file.
+
+```json
+"catalog-venia": {
+    "type": "vcs",
+        "url": "https://github.com/PMET-public/module-catalog-sample-data-venia"
+
+},
+  "configurable-venia": {
+      "type": "vcs",
+          "url": "https://github.com/PMET-public/module-configurable-sample-data-venia"
+
+  },
+  "customer-venia": {
+      "type": "vcs",
+          "url": "https://github.com/PMET-public/module-customer-sample-data-venia"
+
+  },
+  "sales-venia": {
+      "type": "vcs",
+          "url": "https://github.com/PMET-public/module-sales-sample-data-venia"
+
+  },
+  "tax-venia": {
+      "type": "vcs",
+          "url": "https://github.com/PMET-public/module-tax-sample-data-venia"
+
+  },
+  "media-venia": {
+      "type": "vcs",
+          "url": "https://github.com/PMET-public/sample-data-media-venia"
+
+  },
+```
+
+### Require the sample data modules
+
+Add the sample data modules to the `require` section of the `composer.json` file to install them into Magento.
+
+```json
+"magento/module-catalog-sample-data-venia": "dev-master",
+"magento/module-configurable-sample-data-venia": "dev-master",
+"magento/module-customer-sample-data-venia": "dev-master",
+"magento/module-sales-sample-data-venia": "dev-master",
+"magento/module-tax-sample-data-venia": "dev-master",
+"magento/sample-data-media-venia": "dev-master",
+```
+
+## Modify `.gitignore` file
+
+Add the following frontend dependency entries to the `.gitignore` file to track them in your repository.
+
+```text
+!package.json
+!yarn.lock
```

Comment:
I've never seen the `!` in a `.gitignore`, what does it do?

Reply:
It's a negation. So these specific files are guaranteed to be tracked.

Reply:
Isn't the same thing achieved if we just leave them out of `.gitignore`? Maybe we could drop this whole step?

Reply:
I think @bbatsche included this to make the point that these files are important and should be tracked.

Reply:
If you take a look at the default `.gitignore` we use for cloud (https://github.com/magento/magento-cloud/blob/master/.gitignore) you'll see that it is kind of an inverse ignore file. By default, all files are excluded from git, and then specific files or directories are whitelisted back in